### ISSUE TITLE
Applying visual effects

### ISSUE THREAD

**Issue by chrishtr** [mathml-refresh/mathml#179]
Has there been any discussion of what to do when transform, filter, opacity, blend-mode, clip-path, clip, etc are applied to a MathML element?

This is very related to the containing block and stacking context discussion (e.g. #16).

**Comment by fred-wang** [comment]
@chrishtr Hi, sorry this was posted during vacations and I missed it. I don't think we had any discussion in the CG but my personal opinion is that we should probably support them if possible. I think @bfgeek had already mentioned them somewhere.

**Comment by fred-wang** [comment]
I didn't mention it here, but in order to address #52 a section was added saying that basically all MathML elements generate a stacking context: https://mathml-refresh.github.io/mathml-core/#stacking-contexts

I'm not sure if these effects are really important for math formulas, but I wouldn't surprised if someone wants to use them (at least opacity) and if they are already supported by webkit or gecko. IIUC, @bfgeek said it's easier to just allow these properties that to disable them.

Also  I don't know if that's included in this issue, but at least I used linear-gradient on an mrow element some years ago for https://fred-wang.github.io/MathFonts/STIX/index.html and it's working in all browsers.

**Comment by rwlbuis** [comment]
Going to experiment... Expectation is that nothing/not much needs to be added to the specification about this, but we need tests for sure.

**Comment by NSoiffer** [comment]
For the record, from the MathML Core meeting today:
NS: Seems like we need to write tests and see what the implementation is doing. Do they just flow through or do we need to have the UA stylesheet for math reset some of them?
BM: Seems like we need more abstract language so we don’t have to list every potential interaction. Probably requires someone with deep knowledge of the web to know which things fall into what categories… if there are different categories of effects.
BK: we need to know where it is special, such as creating a new stacking context.

Bottom line is that people generally felt they should work unless there is some reason they shouldn't work such as messing up the implementation.

**Comment by rwlbuis** [comment]
So far my experiments showed up no problems. Still need to add a test for opacity I think.

**Comment by NSoiffer** [comment]
Based on comment by @rwlbuis at the core meeting today, there don't appear to be any problems with supporting transforms. Some more tests probably need to be added and Rob will check with google about what they think should be tested.

**Comment by rwlbuis** [comment]
@chrishtr Several subtests landed here:
https://github.com/web-platform-tests/wpt/commit/9b98f401e54a7acf9f2dc889318ee7c5d56c1263
can you think of more visual effects to test? As mentioned before I don't think we want tests for all ~11 css filters...

**Comment by davidcarlisle** [comment]
Should ths issue be moved to mathml-core?

**Comment by NSoiffer** [comment]
Yes
